### PR TITLE
Upgrade to stellar-sdk 2.10 and SEP-10 v3.0

### PR DESCRIPTION
**Pull request by JakeUrban**
Upgrades to the Python SDK version with SEP-10 v3.0 support and adjusts the SEP-10 implementation in accordance with the protocol changes.

Adds new environment variables:
- `SEP10_HOME_DOMAINS`: a list of home domains Polaris will use when validating client requests and SEP-10 challenges

Updates Documentation:
- Provides a more explicit and organized list of the environment variables used by Polaris and their definitions

Changes SEP10 Logic:
- Rejects client requests containing a `home_domain` parameter that is not included in `SEP10_HOME_DOMAINS`
- Uses the first entry in `SEP10_HOME_DOMAINS` for constructing challenges instead of `HOST_URL.hostname`

### CODE REVIEW

**Comment by JakeUrban** [comment]
@ronaldoy feel free to take a look. This didn't require another integration, just a new env variable.

**Comment by ronaldoy** [comment]
@JakeUrban Works well!

**Comment by ronaldoy** [comment]
@JakeUrban It works well.

**Review by Zagan202 — CHANGES_REQUESTED** [review]
Passed all the tests when I updated my `django-polaris/polaris/.env` to include 
```
SEP10_HOME_DOMAINS=localhost,testanchor.stellar.org,example.com
```
(I made it in the format shown in the docs ``SEP10_HOME_DOMAINS=testanchor.stellar.org,example.com``)

**A few things**

1. We should update `django-polaris/polaris/.env.example` and `django-polaris/example/.env.example` to include SEP10_HOME_DOMAINS
2. Update the contributing section to include SEP10_HOME_DOMAINS parms

**Review by ronaldoy — APPROVED** [review]

**Comment by JakeUrban** [comment]
@Zagan202 should be ready for your review again

File: example/.env.example [hunk 1]
```diff
@@ -1,24 +1,11 @@
...
-# HORIZON_URI can point to a custom Horizon URI. It currently points
-# to the testnet URL.
 HORIZON_URI="https://horizon-testnet.stellar.org/"
-
 SERVER_JWT_KEY="secret"
-
 HOST_URL="https://example.com"
-
-# Change this to False or remove when running on HTTPS
 LOCAL_MODE=True
```
Comment (Zagan202) [inline example/.env.example:11]:
```suggestion
LOCAL_MODE=True
SEP10_HOME_DOMAINS="example.com"
```

**Review by Zagan202 — APPROVED** [review]
One minor suggestion but LGTM

**Pull request merged**